File: code/python/postgresql.py
import psycopg2                     # pip install psycopg2
from dotenv import load_dotenv      # pip install python-dotenv
import os

load_dotenv()


class PostgreSql():

    def __init__(self) -> None:

        self.user = os.getenv("USER")
        self.password = os.getenv("PASSWORD")
        self.database = os.getenv("DATABASE")
        self.host = os.getenv("HOST")

    def connect_to_database(self):
        """
        Establishes and active connection to the database and returns to connection
        """
        conn = psycopg2.connect(
            database=self.database,
            host=self.host,
            user=self.user,
            password=self.password
        )

        if conn:
            return conn
        else:
            raise Exception("Database connection failed")

    def get_all_users(self):
        '''
        This example will get all users from the database
        '''

        conn = self.connect_to_database()
        cur = conn.cursor()
        try:
            cur.execute('''
                    SELECT * FROM users;
            ''')
            query = cur.fetchall()
            conn.commit()
            conn.close()
            return query
        except Exception as e:
            print(e)
            conn.rollback()
            conn.close()

    def get_user(self, user):
        '''
        This example will use f-string to get a user
        '''
        conn = self.connect_to_database()
        cur = conn.cursor()
        try:
            cur.execute(f'''
                SELECT * FROM users
                WHERE username = '{user}';
            ''')
            query = cur.fetchone()
            conn.commit()
            conn.close()
            return query
        except Exception as e:
            print(e)
            conn.rollback()
            conn.close()


if __name__ == '__main__':
    db = PostgreSql()
    print(db.get_all_users())
    print(db.get_user('stoffel'))

File: code/javascript/postgresql.js
"use strict";

const { Client } = require("pg");               // npm i pg
require("dotenv").config({path: '../.env'});    // npm i dotenv

class Database {
  constructor() {
    (this.database = process.env.DATABASE),
      (this.host = process.env.HOST),
      (this.user = process.env.USER),
      (this.password = process.env.PASSWORD);
  }

  connectToDatabase = () => {
    /* Establish a connection to the database
       and return a client object */
    const client = new Client({
      host: this.host,
      port: 5432,
      database: this.database,
      user: this.username,
      password: this.password,
      idle_in_transaction_session_timeout: 3,
    });
    return client;
  };

  getUser = async (argument) => {
    const client = this.connectToDatabase();
    await client.connect();
    try {
      const query = await client.query(`SELECT * FROM users
                                        WHERE username = '${argument}';`);
      await client.end();
      return query.rows;
    } catch (err) {
      console.log(err);
    }
  };
}

const test = new Database();
test.getUser("stoffel").then(res => console.log(res));
